File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_library/lib/src/services/library_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/library_models.dart';

/// Library service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class LibraryService {
  const LibraryService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'library';

  LibraryPageResult<Never> empty() =>
      const LibraryPageResult<Never>(items: <Never>[]);
}

File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-library/src/services/libraryService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { LibraryPageResult } from "../types/libraryModels.js";

/**
 * Library service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface LibraryService {
  readonly capability: "library";
}

export function createLibraryService(client: AppstoreAppClient): LibraryService {
  void client;
  return { capability: "library" };
}

export function emptyLibraryResult(): LibraryPageResult<never> {
  return { items: [] };
}
